hotxlfp/grammarparser/lexer.py

```diff
@@ -38,6 +39,12 @@
 )
 
 
+
+def t_SCIENTIFIC_NOTATION_E(t):
+    r'(?<=\d)\s*e\s*(?=\d)'  # Lookahead to not confuse with variables and functions
```

Comment:
I'm not sure if this lookahead / lookbehind should be needed. Because I would have thought that the if the string could not parse the e as a scientific notation e, it should fallback to trying to parse it as a variable. 

But when I tested, this doesn't work. I tried both making the scientific notation e have higher and lower priority than the variable tokens. But I might just not understand lexing / parsing enough.

This solution with the regex seems to work though. This is also similar to what I did for functions.